app/init.php

```diff
@@ -791,205 +233,172 @@ function (mixed $value) {
-                            'mariadb' => new MariaDB($resource()),
-                            'mysql' => new MySQL($resource()),
-                            default => null
-                        };
-
-                        $adapter->setDatabase($dsn->getPath());
-                        break;
-                    case 'pubsub':
-                        $adapter = $resource();
-                        break;
-                    case 'queue':
-                        $adapter = match ($dsn->getScheme()) {
-                            'redis' => new Queue\Connection\Redis($dsn->getHost(), $dsn->getPort()),
-                            default => null
-                        };
+            foreach ($dsns as &$dsn) {
+                $dsn = explode('=', $dsn);
+                $name = ($multiple) ? $key . '_' . $dsn[0] : $key;
+                $config[] = $name;
+                $dsn = $dsn[1] ?? '';
+
+                if (empty($dsn)) {
+                    throw new Exception(Exception::GENERAL_SERVER_ERROR, "Missing value for DSN connection in {$key}");
+                }
+
+                $dsn = new DSN($dsn);
+                $dsnHost = $dsn->getHost();
+                $dsnPort = $dsn->getPort();
+                $dsnUser = $dsn->getUser();
+                $dsnPass = $dsn->getPassword();
+                $dsnScheme = $dsn->getScheme();
+                $dsnDatabase = $dsn->getPath();
+
+                if (!in_array($dsnScheme, $schemes)) {
+                    throw new Exception(Exception::GENERAL_SERVER_ERROR, "Invalid console database scheme");
+                }
+
+                /**
+                 * Get Resource
+                 *
+                 * Creation could be reused accross connection types like database, cache, queue, etc.
+                 *
+                 * Resource assignment to an adapter will happen below.
+                 */
+                switch ($dsnScheme) {
+                    case 'mysql':
+                    case 'mariadb':
+                        $pool = new PDOPool(
+                            (new PDOConfig())
+                                ->withHost($dsnHost)
+                                ->withPort($dsnPort)
+                                ->withDbName($dsnDatabase)
+                                ->withCharset('utf8mb4')
+                                ->withUsername($dsnUser)
+                                ->withPassword($dsnPass)
+                                ->withOptions([
+                                    // No need to set PDO::ATTR_ERRMODE it is overwitten in PDOProxy
+                                    // PDO::ATTR_TIMEOUT => 3, // Seconds
+                                    // PDO::ATTR_PERSISTENT => true,
+                                    PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC,
+                                    PDO::ATTR_EMULATE_PREPARES => true,
+                                    PDO::ATTR_STRINGIFY_FETCHES => true,
```

Comment:
previously these were present
PDO::ATTR_TIMEOUT => 3, // Seconds
PDO::ATTR_PERSISTENT => true,

and PDO::MYSQL_ATTR_USE_BUFFERED_QUERY => true, was not there.. 
are these changes required?